notNeededPackages.json

```diff
@@ -1725,6 +1725,10 @@
             "libraryName": "global-dirs",
             "asOfVersion": "1.0.0"
         },
+        "glob": {
+            "libraryName": "glob",
+            "asOfVersion": "9.0.0"
+        },
```

Comment:
Given the problems I've seen with deprecating widely used packages, I think it would be better to keep `glob` out of notNeededPackages and therefore skip the publication of an empty, deprecated version. It'll be less obvious that people should stop installing `@types/glob`, but more likely to avoid problems when people install something that currently depends on `@types/glob@*`, which in turn depends on `@types/node@*`.

Reply:
Is this likely to be true? A sourcegraph search shows that this might be somewhat rare:

Reply:
Basically, who's using `"*"` for `@types/glob` that isn't also using `"*"` for `glob` itself? From my skimming of the above, it seems like either people pin both or pin neither.

Reply:
Oh, duh. I forgot about that from last time. Types packages star depend on each other. https://unpkg.com/browse/@types/dos2unix@1.1.1/package.json

Reply:
If we keep packages like this in DefinatelyTyped, we will not move the landscape forward. There is a documented procedure on what to do when types move into their own repository no?

https://github.com/DefinitelyTyped/DefinitelyTyped#removing-a-package

> When a package [bundles](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) its own types, types should be removed from Definitely Typed to avoid confusion.

Reply:
I agree: the types should be removed. But I don't think they should be entered in notNeededPackages.json. That publishes a final version with no contents, and is then deprecated. However, the preceding discussion explains why that's a bad idea for widely-used packages, which tend to be used in the middle of a large dependency graph.

Either way, if you install a `glob` with types and `@types/glob`, Typescript will ignore `@types/glob`, so `glob` will move forward regardless of what happens on DT.

Reply:
> Either way, if you install a glob with types and @types/glob, Typescript will ignore @types/glob, so glob will move forward regardless of what happens on DT.

This is not nessesarily good no? I am experiencing problems, read my comment for context:

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64806#issuecomment-1477676504

It breaks the build for my project

Reply:
I think I understand the problem you explained in #64806. I'll comment over there to make sure.

For this PR, it won't help or hurt your problem to add `@types/glob` to notNeededPackage.json, and adding it has the potential to hurt other people who need a package that depends on `@types/glob`.

Reply:
Right, so should we close this PR?

I don't understand, should we not follow the guidelines?

> When a package bundles its own types, types should be removed from Definitely Typed to avoid confusion.

Reply:
No, we're saying that you can just delete the glob folder and not update notNeededPackage.json.

Reply:
I see, I have removed this update

Is there anything else to do for this PR?

Reply:
No, though the CI is failing which I think is now showing that some packages depend on these types?

Reply:
Yes, they do, should I modify all of those types? It is a popular package

Reply:
Yes, they should be updated in whatever method seems right. Could be pinning an old version, could be just depending on the new one.

Reply:
I am almost done, but the CI is failing on an unrelated package. It seems like it cannot `npm install` or something